From 99ece7b57d8a928bdf39756a60d9369f2e05d473 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Tue, 24 Jul 2012 09:47:33 +0200
Subject: [PATCH] rules: avoid mounting raid devices too early

/dev/md0 appears as soon as the first component of the raid array is
added by incremental assembly rules. This is too early for systemd to
attempt to mount the device. The device should be considered plugged
after the raid array becomes active.

https://bugzilla.redhat.com/show_bug.cgi?id=767561
(cherry picked from commit a96e4839d5ba1b2cbc75d9a998b5f8ccfbb39544)
---
 src/99-systemd.rules.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/99-systemd.rules.in b/src/99-systemd.rules.in
index 8cc7523..ff92ebf 100644
--- a/src/99-systemd.rules.in
+++ b/src/99-systemd.rules.in
@@ -19,6 +19,9 @@ SUBSYSTEM=="block", KERNEL!="ram*|loop*", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=
 
 SUBSYSTEM=="block", KERNEL!="ram*|loop*", ENV{DM_UUID}=="CRYPT-*", ENV{ID_PART_TABLE_TYPE}=="", ENV{ID_FS_USAGE}=="", ENV{SYSTEMD_READY}="0"
 
+# Ignore raid devices that are not yet assembled and started
+SUBSYSTEM=="block", KERNEL=="md*", ATTR{md/array_state}=="|clear|inactive", ENV{SYSTEMD_READY}="0"
+
 # We need a hardware independent way to identify network devices. We
 # use the /sys/subsystem path for this. Current vanilla kernels don't
 # actually support that hierarchy right now, however upcoming kernels
